File: core/src/ua/syt0r/actors/ui/SolidColorActor.java
package ua.syt0r.actors.ui;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.NinePatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.scenes.scene2d.Actor;

public class SolidColorActor extends Actor{

    private NinePatch ninePatch;
    private Color defaultColor = Color.valueOf("5e5e5e");

    public SolidColorActor(TextureRegion textureRegion, float width, float height){
        setSize(width,height);
        ninePatch = new NinePatch(textureRegion, defaultColor);
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        Color color = getColor();
        batch.setColor(color.r, color.g, color.b, color.a * parentAlpha);
        ninePatch.draw(batch,getX(),getY(),getOriginX(),getOriginY(),getWidth(),getHeight(),getScaleX(),getScaleY(),getRotation());
        batch.setColor(color.r, color.g, color.b, 1f);
    }
}

File: core/src/ua/syt0r/actors/ui/SolidColorActor.kt
package ua.syt0r.actors.ui

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.Pixmap
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.Actor

class SolidColorActor(
    x: Float,
    y: Float,
    width: Float,
    height: Float,
    color: Color = Color.valueOf("5e5e5e")
) : Actor() {

    private val texture: Texture

    init {
        setBounds(x, y, width, height)
        val pixmap = Pixmap(1, 1, Pixmap.Format.RGB888)
        pixmap.setColor(color)
        pixmap.fill()
        texture = Texture(pixmap)
        pixmap.dispose()
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        batch.draw(texture, x, y, width, height)
    }

}